src/main/java/org/netcracker/educationcenter/elasticsearch/Connection.java

```diff
@@ -0,0 +1,51 @@
+package org.netcracker.educationcenter.elasticsearch;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import java.io.IOException;
+
+/**
+ * This class establishes a connection with Elasticsearch Database using rest client
+ * to perform requests based on the provided builder.
+ *
+ * @author Mikhail Savin
+ */
+public class Connection {
+
+    /**
+     * High-level client instance to establish connection
+     */
+    private RestHighLevelClient restHighLevelClient;
+
+    /**
+     * @return High-level client instance
+     */
+    public RestHighLevelClient getRestHighLevelClient() {
+        return restHighLevelClient;
+    }
+
+    /**
+     * High-level client internally creates the low-level client
+     * (which maintains a pool of connections and starts some threads) to perform requests.
+     * Note that we need both ports to be added via builder.
+     */
+    public synchronized void makeConnection() {
+        if (restHighLevelClient == null) {
+            restHighLevelClient = new RestHighLevelClient(
+                    RestClient.builder(
+                            new HttpHost("142.93.122.167", 9200, "http"),
+                            new HttpHost("142.93.122.167", 9201, "http")));
+        }
+    }
+
+    /**
+     * Closes the client to free resources.
+     * @throws IOException if something wrong with client instance(for example: it doesn't exist)
+     */
+    public synchronized void closeConnection() throws IOException {
```

Comment:
Maybe class should implement Closeable or Autocloseable interface

Reply:
```
@Override
    public void close() throws Exception {
        restHighLevelClient.close();
        restHighLevelClient = null;
    }
```

Now it implements AutoCloseable interface. Maybe its better to invoke makeConnection() method in constructor so we don't have to do it manually by connection.makeConnection()?

Reply:
public Connection() {
        this.property = new Properties();
        makeConnection();
    }